Galaxus.Functional/Linq/OptionLinqExtensions.cs

```diff
@@ -123,8 +123,23 @@ public static Option<TSource> SingleOrNone<TSource>(this IEnumerable<TSource> so
                 .Select(Option<TSource>.Some)
                 .SingleOrDefault();
 
+        /// <summary>
+        /// Checks a Dictionary if it contains a value for a given key. If so this value is wraped in a Option. Else <see cref="Option{T}.None"/> is returned.
+        /// </summary>
+        /// <param name="dictionary">The dictionary in quesion</param>
+        /// <param name="key">The key to check if a value exists</param>
+        /// <returns>The Value of the Dictionary or None</returns>
+        public static Option<TSome> GetValueOrNone<TKey, TSome>(
+            this IReadOnlyDictionary<TKey, TSome> dictionary,
```

Comment:
I wonder - do we want to limit ourselves to read only dictionaries? I think allowing any `IDictionary` would be more fitting.

(And by the way - there's a typo in the xmldoc.. "The dictionary in **quesion**" :D

Reply:
The problem here, is that IReadOnlyDictionary and IDictionary don't share an interface for the method TryGetValue. But I think I found a solution for that.

Reply:
It's not even about `TryGetValue`, their common interface is just `IEnumerable<KeyValuePair<TKey, TValue>>`. That's a bit of a pain...

I like your solution, it's very much how most of LINQ is done (e.g. `ToList()` - "if its already a list, do this; if its not, do that"). But I wouldn't throw if the "dictionary" is not a dictionary per se. It's an `IEnumerable<T>`, so we can leverage our other extensions.

What about something like this?

```csharp
public static Option<TSome> GetValueOrNone<TKey, TSome>(this IEnumerable<KeyValuePair<TKey, TSome>> source, TKey key)
{
    switch (source)
    {
        case IDictionary<TKey, TSome> dictionary:
            return dictionary.ContainsKey(key) ? Option<TSome>.Some(dictionary[key]) : Option<TSome>.None;
        case IReadOnlyDictionary<TKey, TSome> readOnlyDictionary:
            return readOnlyDictionary.ContainsKey(key) ? Option<TSome>.Some(readOnlyDictionary[key]) : Option<TSome>.None;
        default:
            return source.SingleOrNone(kvp => kvp.Key.Equals(key)).Map(kvp => kvp.Value);
    }
}
```